corehq/apps/locations/tests/test_location_queries.py

```diff
@@ -191,30 +192,39 @@ class TestFilterByUserInput(LocationHierarchyTestCase):
         ])
     ]
 
-    def test_path_queries(self):
-        for querystring, expected in [
-            ('', SQLLocation.objects.filter(domain=self.domain)
-                                    .values_list('name', flat=True)),
-            ('Suff', ['Suffolk']),
-            ('Suffolk', ['Suffolk']),
-            ('Cambridge', ['Cambridge', 'Cambridge']),
-            ('Massachusetts/Cambridge', ['Cambridge']),
-            ('"Copycat"/Cambridge', []),
-            ('"Middlesex"/Cambridge', ['Cambridge']),
-            ('"Middlesex/Cambridge', ['Cambridge', 'Cambridge']),
-            ('"Middlesex"/Somerville', ['Somerville', 'Evil Somerville']),
-            ('"Middlesex"/"Somer', ['Somerville', 'Evil Somerville']),
-            ('"Middlesex"/"Somerville"', ['Somerville']),
-            ('mass/mid/Som', ['Somerville', 'Evil Somerville']),
-            ('Middl', ['Middlesex', 'Evil Middlesex']),
-            ('Middl/', ['Cambridge', 'Somerville', 'Evil Somerville', 'Cambridge', 'Somerville']),
-            ('Middl/camb', ['Cambridge', 'Cambridge']),
-        ]:
-            actual = list(SQLLocation.objects
-                          .filter_by_user_input(self.domain, querystring)
-                          .values_list('name', flat=True))
-            error_msg = f"\nExpected '{querystring}' to yield\n{expected}\nbut got\n{actual}"
-            self.assertItemsEqual(actual, expected, error_msg)
+    def assert_query_has_results(self, querystring, expected):
+        actual = list(SQLLocation.objects
+                      .filter_by_user_input(self.domain, querystring)
+                      .values_list('name', flat=True))
+        error_msg = f"\nExpected '{querystring}' to yield\n{expected}\nbut got\n{actual}"
+        self.assertItemsEqual(actual, expected, error_msg)
+
+    def test_empty_path_query(self):
+        all_locations = (
+            SQLLocation.objects
+            .filter(domain=self.domain)
+            .values_list('name', flat=True)
+        )
+        self.assert_query_has_results('', all_locations)
+
+@generate_cases([  # noqa: E302
+    ('Suff', ['Suffolk']),
+    ('Suffolk', ['Suffolk']),
+    ('Cambridge', ['Cambridge', 'Cambridge']),
+    ('Massachusetts/Cambridge', ['Cambridge']),
+    ('"Copycat"/Cambridge', []),
+    ('"Middlesex"/Cambridge', ['Cambridge']),
+    ('"Middlesex/Cambridge', ['Cambridge', 'Cambridge']),
+    ('"Middlesex"/Somerville', ['Somerville', 'Evil Somerville']),
+    ('"Middlesex"/"Somer', ['Somerville', 'Evil Somerville']),
+    ('"Middlesex"/"Somerville"', ['Somerville']),
+    ('mass/mid/Som', ['Somerville', 'Evil Somerville']),
+    ('Middl', ['Middlesex', 'Evil Middlesex']),
+    ('Middl/', ['Cambridge', 'Somerville', 'Evil Somerville', 'Cambridge', 'Somerville']),
+    ('Middl/camb', ['Cambridge', 'Cambridge']),
+], TestFilterByUserInput)
+def test_path_query(self, querystring, expected):
+    self.assert_query_has_results(querystring, expected)
```

Comment:
Not at all blocking, but just curious to hear more about the rationale for putting assertions in helpers. I like keeping asserts simple, even if that means have more code/duplicate code since it is easy to see what is being asserted in the test, rather than having to go find the helper. That is obviously a nit, and more of an issue when reviewing PRs than working with the tests.

Reply:
The thought of duplicating the logic in that assertion in each of the new test cases seems worse to me than putting it in a concise assert method.

Previously it was the body of a for-loop where each loop was effectively a separate test case, except not actually. Do you think that was better?

Reply:
Yeah sorry I wasn't clear about which duplication I was referring to. You could just have the logic that is in `assert_query_has_results` in `test_path_query` directly. Sure, there might be other tests that do a similar assertion, but that is the tradeoff with duplication that I'm ok with. Recognize that this is just a personal preference though. I am a fan of using `generate_cases` 👍🏻

Reply:
We were on the same page with regard to what you were suggesting, but I think we disagree on the best way to write it. I don't like the idea of duplicating the body of `assert_query_has_results` in multiple tests; I like the helper function.

Reply:
Yeah so I think we were not on the same page. The helper assert is what I was commenting on in the first place 😅 with the main rationale being it makes reviewing changes to tests a little unclear because I have to go find the helper to understand what is being asserted (take the last commit in this PR for example). But again, happy to agree to disagree!

Reply:
For what it's worth I'm on team "minimize duplication in tests" in general too.  Some coding book I read years ago preached that tests merit the same considerations you'd give to production code, which resonates with me.  I've worked in test suites with a lot of duplication before and found it hard to make cross-cutting updates.  At the extremes they are verbose and hard to read (like if each case here was its own test without a helper, which I know no one is advocating for, but which we do elsewhere).

That said, [Locality of Behavior](https://htmx.org/essays/locality-of-behaviour/) is a super important consideration for tests, though that's not violated if the abstractions are clean and intuitive.